src/frontend/planner_test/tests/testdata/nexmark_watermark.yaml

```diff
@@ -0,0 +1,2221 @@
+        WHERE A.id = B.auction and B.date_time between A.date_time and A.expires
+        GROUP BY A.id, A.seller
+    ) AS Q;
+  planner_error: 'Invalid input syntax: column must appear in the GROUP BY clause or be used in an aggregate function'
+- id: nexmark_q7
+  before:
+  - create_sources
+  sql: |
+    SELECT
+      B.auction,
+      B.price,
+      B.bidder,
+      B.date_time
+    FROM
+      bid B
+    JOIN (
+      SELECT
+        MAX(price) AS maxprice,
+        window_end as date_time
+      FROM
+        TUMBLE(bid, date_time, INTERVAL '10' SECOND)
+      GROUP BY
+        window_end
+    ) B1 ON B.price = B1.maxprice
+    WHERE
+      B.date_time BETWEEN B1.date_time - INTERVAL '10' SECOND
+      AND B1.date_time;
+  batch_plan: |
+    BatchExchange { order: [], dist: Single }
+    └─BatchHashJoin { type: Inner, predicate: $expr4 = max($expr8) AND ($expr5 >= $expr9) AND ($expr5 <= $expr7), output: [$expr2, $expr4, $expr3, $expr5] }
+      ├─BatchExchange { order: [], dist: HashShard($expr4) }
+      | └─BatchProject { exprs: [Field(bid, 0:Int32) as $expr2, Field(bid, 1:Int32) as $expr3, Field(bid, 2:Int32) as $expr4, Field(bid, 5:Int32) as $expr5] }
+      |   └─BatchFilter { predicate: (event_type = 2:Int32) }
+      |     └─BatchProject { exprs: [event_type, person, auction, bid, Case((event_type = 0:Int32), Field(person, 6:Int32), (event_type = 1:Int32), Field(auction, 5:Int32), Field(bid, 5:Int32)) as $expr1, _row_id] }
+      |       └─BatchSource { source: "nexmark", columns: ["event_type", "person", "auction", "bid", "_row_id"], filter: (None, None) }
+      └─BatchExchange { order: [], dist: HashShard(max($expr8)) }
+        └─BatchProject { exprs: [max($expr8), $expr7, ($expr7 - '00:00:10':Interval) as $expr9] }
+          └─BatchHashAgg { group_key: [$expr7], aggs: [max($expr8)] }
+            └─BatchExchange { order: [], dist: HashShard($expr7) }
+              └─BatchProject { exprs: [(TumbleStart(Field(bid, 5:Int32), '00:00:10':Interval) + '00:00:10':Interval) as $expr7, Field(bid, 2:Int32) as $expr8] }
+                └─BatchFilter { predicate: (event_type = 2:Int32) }
+                  └─BatchProject { exprs: [event_type, person, auction, bid, Case((event_type = 0:Int32), Field(person, 6:Int32), (event_type = 1:Int32), Field(auction, 5:Int32), Field(bid, 5:Int32)) as $expr6, _row_id] }
+                    └─BatchSource { source: "nexmark", columns: ["event_type", "person", "auction", "bid", "_row_id"], filter: (None, None) }
+  stream_plan: |
+    StreamMaterialize { columns: [auction, price, bidder, date_time, _row_id(hidden), $expr6(hidden)], stream_key: [_row_id, $expr6, price], pk_columns: [_row_id, $expr6, price], pk_conflict: "NoCheck" }
+    └─StreamProject { exprs: [$expr2, $expr4, $expr3, $expr5, _row_id, $expr6] }
+      └─StreamFilter { predicate: ($expr5 >= $expr7) AND ($expr5 <= $expr6) }
+        └─StreamHashJoin { type: Inner, predicate: $expr4 = max($expr4), output: all }
```

Comment:
Let me take a look.

Reply:
It works if we define `bid` separately. 
```sql

CREATE TABLE bid (
    "auction" BIGINT,
    "bidder" BIGINT,
    "price" BIGINT,
    "channel" VARCHAR,
    "url" VARCHAR,
    "date_time" TIMESTAMP,
    "extra" VARCHAR,
    watermark for date_time as date_time - INTERVAL '1' SECOND
) append only;

explain create materialized view q7 as SELECT
      B.auction,
      B.price,
      B.bidder,
      B.date_time
    FROM
      bid B
    JOIN (
      SELECT
        MAX(price) AS maxprice,
        window_end as date_time
      FROM
        TUMBLE(bid, date_time, INTERVAL '10' SECOND)
      GROUP BY
        window_end
    ) B1 ON B.price = B1.maxprice
    WHERE
      B.date_time BETWEEN B1.date_time - INTERVAL '10' SECOND
      AND B1.date_time;

 StreamMaterialize { columns: [auction, price, bidder, date_time, bid._row_id(hidden), $expr1(hidden)], stream_key: [bid._row_id, $expr1, price], pk_columns: [bid._row_id, $expr1, price], pk_conflict: "NoCheck", watermark_columns: [date_time, $expr1(hidden)] }
 └─StreamIntervalJoin { type: Inner, predicate: bid.price = max(bid.price) AND (bid.date_time >= $expr2) AND (bid.date_time <= $expr1), conditions_to_clean_left_state_table: (bid.date_time >= $expr2), conditions_to_clean_right_state_table: (bid.date_time <= $expr1), output_watermarks: [bid.date_time, $expr1] }
   ├─StreamExchange { dist: HashShard(bid.price) }
   | └─StreamTableScan { table: bid, columns: [auction, bidder, price, date_time, _row_id] }
   └─StreamExchange { dist: HashShard(max(bid.price)) }
     └─StreamProject { exprs: [$expr1, max(bid.price), ($expr1 - '00:00:10':Interval) as $expr2], output_watermarks: [$expr1, $expr2] }
       └─StreamAppendOnlyHashAgg { group_key: [$expr1], aggs: [max(bid.price), count], output_watermarks: [$expr1] }
         └─StreamExchange { dist: HashShard($expr1) }
           └─StreamProject { exprs: [(TumbleStart(bid.date_time, '00:00:10':Interval) + '00:00:10':Interval) as $expr1, bid.price, bid._row_id], output_watermarks: [$expr1] }
             └─StreamTableScan { table: bid, columns: [price, date_time, _row_id] }
```

Reply:
I see, 

but because we have to use the three-in-one source to really control the percentage of all three types of events as the Nexmark specifies,
we cannot use a separate source in the performance test.

Reply:
We must create the VIEW with the generated `datatime` column with watermark instead of the original `datatime`.
```SQL

CREATE VIEW PERSON as 
    select (person).auction, (person).bidder, (person).price, (person).channel, (person).url, (person).extra, 
    datatime
 from nexmark where event_type = 0;
# instead of
    CREATE VIEW PERSON as select (person).* from nexmark where event_type = 0;

```